addons/point_of_sale/models/pos_order.py

```diff
@@ -165,6 +170,10 @@ def _process_order(self, order, draft, existing_order):
 
         return pos_order.id
 
+    def _clean_payment_lines(self):
+        self.ensure_one()
+        order_bank_statement_lines = self.env['pos.payment'].search([('pos_order_id', '=', self.id)])
+        order_bank_statement_lines.unlink()
```

Comment:
```suggestion
        order_payments = self.env['pos.payment'].search([('pos_order_id', '=', self.id)])
        order_payments.unlink()
```

let's take this opportunity to rename the variable.